updater/updater.go

```diff
@@ -234,15 +251,19 @@ func deployApp(client controller.Client, app *ct.App, uri string, updateFn updat
 	return nil
 }
 
-func updateSlugURIs(env map[string]string) bool {
-	uris := map[string]string{
-		"SLUGBUILDER_IMAGE_URI": slugbuilderURI,
-		"SLUGRUNNER_IMAGE_URI":  slugrunnerURI,
+func updateSlugImages(env map[string]string) bool {
+	updates := map[string]string{
+		"SLUGBUILDER_IMAGE_ID": slugBuilder.ID,
+		"SLUGRUNNER_IMAGE_ID":  slugRunner.ID,
+
+		// handle legacy apps which used IMAGE_URI rather than IMAGE_ID
+		"SLUGBUILDER_IMAGE_URI": slugBuilder.URI,
+		"SLUGRUNNER_IMAGE_URI":  slugRunner.URI,
 	}
 	updated := false
-	for key, uri := range uris {
-		if v, ok := env[key]; ok && v != uri {
-			env[key] = uri
+	for key, val := range updates {
+		if v, ok := env[key]; ok && v != val {
+			env[key] = val
```

Comment:
Is this code correct? Don't we want to set the new env vars? Currently it looks like they won't be set if because of the `ok` check.